pkg/vault/client.go

```diff
@@ -197,16 +193,44 @@ func WriteSecret(instanceAddr string, secretPath string, secretData map[string]i
 	}
 }
 
-// read secret from vault
-func ReadSecret(instanceAddr string, secretPath string) *api.Secret {
-	secret, err := getClient(instanceAddr).Logical().Read(secretPath)
+// read secret from vault and return the secret map
+func ReadSecret(instanceAddr, secretPath, engineVersion string) map[string]interface{} {
```

Comment:
Add a comment explaining why KV V2 secret version is not important for this method